crates/astria-core/src/sequencer/v1alpha1/block.rs

```diff
@@ -366,6 +393,25 @@ impl SequencerBlock {
         }
     }
 
+    #[must_use]
+    pub fn filtered_block(self, rollup_ids: Vec<RollupId>) -> Self {
```

Comment:
this should take `&self`? unless you intended for this to take the rollup txs out of `self`?